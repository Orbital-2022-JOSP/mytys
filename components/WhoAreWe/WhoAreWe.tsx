import Link from "next/link"
import { GenericContainer } from '../GenericContainer/GenericContainer';

export const WhoAreWe: React.FC = () => {
    return (
        <GenericContainer>
            <div className="p-8 rounded shadow-xl sm:p-16">
                <div className="flex flex-col lg:flex-row">
                    <div className="mb-6 lg:mb-0 lg:w-1/2 lg:pr-5">
                        <h2 className="font-sans text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl sm:leading-none">
                            What exactly is{" "}
                            <span className="inline-block text-light-blue-accent-400">
                                MYTYS
                            </span>
                        </h2>
                    </div>
                    <div className="lg:w-1/2">
                        <p className="mb-4 text-base text-gray-700">
                            MYTYS is an all-in-one educational webpage that offers a gamified, tuition-like experience as an alternative to conventional, costly tuition.
                            MyTYS is created by <span className="text-light-blue-accent-400">students</span>, for <span className="text-light-blue-accent-400">students</span>
                        </p>
                        <Link href={"/"}>
                            <a
                                aria-label=""
                                className="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
                            >
                                Learn more
                            </a>
                        </Link>
                    </div>
                </div>
            </div>
        </GenericContainer>
    )
}