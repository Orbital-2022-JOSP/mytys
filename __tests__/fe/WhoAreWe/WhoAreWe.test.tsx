import { render } from '@testing-library/react';
import { WhoAreWe } from '../../../components/WhoAreWe/WhoAreWe';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Who Are We Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<WhoAreWe />);

            expect(container).toMatchSnapshot()
        })
})

