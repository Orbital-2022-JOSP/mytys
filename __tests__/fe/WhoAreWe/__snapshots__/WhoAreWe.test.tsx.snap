// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Who Are We Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="p-8 rounded shadow-xl sm:p-16"
    >
      <div
        class="flex flex-col lg:flex-row"
      >
        <div
          class="mb-6 lg:mb-0 lg:w-1/2 lg:pr-5"
        >
          <h2
            class="font-sans text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl sm:leading-none"
          >
            What exactly is
             
            <span
              class="inline-block text-light-blue-accent-400"
            >
              MYTYS
            </span>
          </h2>
        </div>
        <div
          class="lg:w-1/2"
        >
          <p
            class="mb-4 text-base text-gray-700"
          >
            MYTYS is an all-in-one educational webpage that offers a gamified, tuition-like experience as an alternative to conventional, costly tuition. MyTYS is created by 
            <span
              class="text-light-blue-accent-400"
            >
              students
            </span>
            , for 
            <span
              class="text-light-blue-accent-400"
            >
              students
            </span>
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
            href="/"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
